### PR TITLE
Move pyxet into xet-core

### PR DESCRIPTION
**Pull request by hoytak**
Pyxet development is connected to xet-core development, and having a separate python repository has introduced unneeded friction in the development process.  In addition, some backend errors were introduced recently that were only caught in the pyxet unit tests, so running these as part of the regular xet-core development will make the build process more robust.   

This PR takes a source drop of pyxet, switches the dependency path of pyxet to be the local directory (through libxet), and merges the github actions of the two repositories.

### CODE REVIEW

File: .github/workflows/commit.yml [hunk 1]
```diff
@@ -140,16 +140,15 @@ jobs:
         if: runner.os != 'macOS'
         run: |
           cd python/pyxet
-          maturin build -r
+          maturin build --debug
```
Comment (seanses) [inline .github/workflows/commit.yml:143]: I think no option specified means debug build. cargo apparently complains about "error: unexpected argument '--debug' found"

File: .github/workflows/commit.yml [hunk 2]
```diff
@@ -140,16 +140,15 @@ jobs:
...
       - name: Build with maturin (macOS)
         if: runner.os == 'macOS'
         run: |
           brew install make
           export PATH="/usr/local/opt/make/libexec/gnubin:$PATH"
           export CXXFLAGS="-stdlib=libc++"
           cd python/pyxet
-          rustup target add aarch64-apple-darwin x86_64-apple-darwin 
-          maturin build --target universal2-apple-darwin -r --features openssl_vendored 
+          maturin build --debug --features openssl_vendored
```
Comment (seanses) [inline .github/workflows/commit.yml:151]: Same as above.

**Review by seanses — APPROVED** [review]
I'm 👍 with this, given that we are going to install git-xet and xet cli together we should also release them in lockstep. May need to confirm with product team over branding and marketing questions.

**Comment by seanses** [comment]
We'll also need to copy over the action secrets and env variables to run the pyxet tests.

**Comment by hoytak** [comment]
> We'll also need to copy over the action secrets and env variables to run the pyxet tests.

I believe these variables are account wide, so we should be fine.

File: .github/workflows/commit.yml (hunk 2 above)
Comment (hoytak) [reply .github/workflows/commit.yml]: Yes, typo!

**Comment by hoytak** [comment]
Vetoed by @ylow, closing.

**Pull request closed without merging**